src/meta/src/barrier/progress.rs

```diff
@@ -16,20 +16,25 @@ use std::collections::hash_map::Entry;
 use std::collections::HashMap;
 use std::sync::Arc;
 
+use risingwave_common::catalog::TableId;
 use risingwave_common::util::epoch::Epoch;
+use risingwave_pb::ddl_service::DdlProgress;
+use risingwave_pb::hummock::HummockVersionStats;
 use risingwave_pb::stream_service::barrier_complete_response::CreateMviewProgress;
 
 use super::command::CommandContext;
 use super::notifier::Notifier;
+use crate::barrier::Command;
 use crate::model::ActorId;
 use crate::storage::MetaStore;
 
 type CreateMviewEpoch = Epoch;
+type ConsumedRows = u64;
 
 #[derive(Clone, Copy)]
 enum ChainState {
     ConsumingSnapshot,
-    ConsumingUpstream(Epoch),
+    ConsumingUpstream(Epoch, ConsumedRows),
     Done,
```

Comment:
This is originally designed for the rearranged-chain, and it seems not much consistent with the current backfill logic. For example, `ConsumingSnapshot` is never used. We can clean it up in the future if unnecessary. 😄